jffs2/libjffs2.c

```diff
@@ -58,17 +44,23 @@ struct inode *jffs2_srv_get(jffs2_partition_t *p, oid_t *oid)
 }
 
 
-static int jffs2_srv_lookup(jffs2_partition_t *p, oid_t *dir, const char *name, oid_t *res, oid_t *dev, char *lnk, int lnksz)
+static int libjffs2_lookup(void *info, oid_t *dir, const char *name, oid_t *res, oid_t *dev, char *lnk, int lnksz)
 {
 	struct dentry *dentry, *dtemp;
 	struct inode *inode = NULL;
 	int len = 0;
 	char *end;
+	jffs2_partition_t *p;
+
+	if (info == NULL)
+		return -EINVAL;
+
+	p = (jffs2_partition_t *)info;
```

Comment:
Why? Can't we just accept `jffs2_partition_t *info` as a parameter?

Reply:
Ok, I think I understand, this is API, and we cannot export this type. If that's right please just resolve this remark

Reply:
It is a generic interface used by the other filesystems as well.